rule/dot-imports.go

```diff
@@ -32,15 +40,47 @@ func (*DotImportsRule) Name() string {
 	return "dot-imports"
 }
 
+func (r *DotImportsRule) configure(arguments lint.Arguments) {
+	r.Lock()
+	defer r.Unlock()
+
+	if r.allowedPackages == nil {
+		r.allowedPackages = make(allowPackages)
+		if len(arguments) > 0 {
```

Comment:
To reduce cognitive complexity, I'd convert here and in other possible cases to guard conditions  (early return pattern).